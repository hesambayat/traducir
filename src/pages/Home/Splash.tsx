import { useEffect, useState } from 'react'

const Splash = () => {
  const [className, setClassName] = useState('splash')

  useEffect(() => {
    const emit = setTimeout(() => {
      setClassName('splash splash--out')
    }, 1200)

    return () => clearTimeout(emit)
  }, [setClassName])

  return (
    <div className={className} />
  )
}

export default Splash
