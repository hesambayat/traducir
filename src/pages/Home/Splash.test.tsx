import { render } from '@testing-library/react'
import Splash from './Splash'

describe('renders home page feature / splash', () => {
  const { asFragment } = render(<Splash />)
  const dom = asFragment()
  test('should match the snapshot', () => {
    expect(dom).toMatchSnapshot()
  })
})
