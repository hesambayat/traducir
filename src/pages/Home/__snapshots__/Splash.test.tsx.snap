// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders home page feature / splash should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="splash"
  >
    <div
      class="clip"
    >
      <h1>
        Loading... 
        <strong>
          Loading...
        </strong>
         
        <em>
          Loading...
        </em>
      </h1>
      <p>
        Loading... 
        <strong>
          Loading...
        </strong>
         
        <em>
          Loading...
        </em>
      </p>
    </div>
  </div>
</DocumentFragment>
`;
